src/accel/LocalTransporter.cc

```diff
@@ -140,6 +142,9 @@ void LocalTransporter::Flush()
         << "Transporting " << buffer_.size() << " tracks from event "
         << event_id_.unchecked_get() + 1 << " with Celeritas";
```

Comment:
It's actually the opposite... Geant uses 1 based indexing and we translate it to zero, so we have to translate it back. I will add a comment or helper function to clarify this point!

Reply:
Wait, looking at this again I don't see anywhere else we add or subtract 1 from the event ID... I thought this message was off by 1 from the other event counters I was seeing in the HGCal (and maybe demo-geant?) instances, but I could be wrong...

Reply:
> Geant uses 1 based indexing and we translate it to zero
I wasn't sure about that, so I double checked here. Geant4 does start, by default, at event 0. So the off by 1 must be set somewhere else.

Reply:
OK, I figured out what was going on. There's a mismatch between the HepMC3 GenEvent number (1-based) and the Geant4 Event ID (0-based).  Furthermore, because we read the HepMC3 events successively in time, rather than corresponding to the Geant4 event partitioning, the event IDs may not match the Geant4 ID.

Printing both the event ID and the HepMC3 event ID, I see this output from all the different worker threads:
```
[1/2] info: Read 100 primaries from HepMC event ID 1 for Geant4 event 2
[0/2] info: Read 100 primaries from HepMC event ID 2 for Geant4 event 0
[1/2] info: Read 100 primaries from HepMC event ID 3 for Geant4 event 3
[0/2] info: Read 100 primaries from HepMC event ID 4 for Geant4 event 1
[1/2] info: Read 100 primaries from HepMC event ID 5 for Geant4 event 4
[0/2] info: Read 100 primaries from HepMC event ID 6 for Geant4 event 6
[0/2] info: Read 100 primaries from HepMC event ID 7 for Geant4 event 7
[0/2] info: Read 100 primaries from HepMC event ID 8 for Geant4 event 8
[1/2] info: Read 100 primaries from HepMC event ID 9 for Geant4 event 5
[1/2] info: Read 100 primaries from HepMC event ID 10 for Geant4 event 10
[1/2] info: Read 100 primaries from HepMC event ID 11 for Geant4 event 11
[0/2] info: Read   8 primaries from HepMC event ID 12 for Geant4 event 9
```
The HepMC3 events are linear and 1-based, but they don't correspond to the Geant4 event IDs because of multithreading.

@stognini Any suggestions here? Should we be changing the Geant4 event ID to match the hepmc3? Do we need to have a rolling queue of events to match the HepMC3 indices? Are the HepMC3 events always going to be 1-indexed and contiguous?

Good catch @mrguilima .

Reply:
Ideally we'd have the Geant4 output to have matching event numbers. We can do this at `BeginOfEventAction` or, if we want to simplify things, I think adding this to `HepMC3PrimaryGenerator::GeneratePrimaryVertex(G4Event* g4_event)` should work just fine:
```cpp
g4_event->SetEventID(gen_event.event_number());
```
I definitely didn't think too much when implementing the HepMC3 reader for our demo, so that's my fault. But we should match event IDs from a generator input and the Geant4/Celeritas output to allow analyses to keep track of what is producing what. I can see people manually filtering stuff and ending up with non-contiguous values from an input, so matching IDs solves all cases.

Regargind HepMC3 starting from 1: HepMC3 can technically have any numbering scheme, as it's just a matter of calling `HepMC3::GenEvent::set_event_number(n)` when creating the event. So if these are being created by our tiny `hepmc3-generator` app, I can fix this really quickly.

Reply:
Since the HepMC reader isn't used by HGCal maybe we should open up another discussion for that... maybe it needs to be an event generator instead of a vertex generator?? Or maybe we need to load all the events ahead of time  like you originally had it. Either way since it's only used for testing at the moment we can postpone: it's a low priority.

Reply:
Created new issue #643.